ted/src/main/java/frc/robot/subsystems/LEDSubsystem.java

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Crescendo - 2024
+// File: LEDSubsystem.java
+// Intent: Forms a subsystem to control the LEDs
+// ************************************************************
+
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ
+
+package frc.robot.subsystems;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BooleanSupplier;
+import java.util.Iterator;
+
+import frc.robot.common.LEDStateAction;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.common.LEDState;
+import com.ctre.phoenix.led.*;
+import com.ctre.phoenix.led.CANdle.LEDStripType;
+import com.ctre.phoenix.led.CANdle.VBatOutputMode;
+
+public class LEDSubsystem extends SubsystemBase {
+    private final CANdle leds;
+    
+      private final int BUFFER_LENGTH = 144;
+      private int startIdx = 0;
+      private HashMap <LEDState, LEDStateAction> ledStateActions = new HashMap<LEDState, LEDStateAction>();
+      private int blinkPeriodInHertz = 25; // increase this for longer blinks!!!
+      private int blinkCounter = 0;
+      private boolean currentBlinkState = false;
+      private boolean lastBlinkState = false;
+      private LEDState currentLEDState = LEDState.Off;
+
+
+      /**
+       * LEDSubsystem
+       * @param canID can id of the CANdle
+        */
+      public LEDSubsystem(int canID) {
+        this.leds = new CANdle(canID); // initialization of the AdressableLED
+        CANdleConfiguration configAll = new CANdleConfiguration();
+        configAll.stripType = LEDStripType.RGB;
+        configAll.brightnessScalar = 0.5;
+        configAll.vBatOutputMode = VBatOutputMode.Modulated;
+        this.leds.configAllSettings(configAll, 100);
+      }
+
+    public void configLedType(LEDStripType type) { 
```

Comment:
I don't think it makes sense for this to be a public method for this class. This is not something we will change on the fly because it's dictated by the LED strip type that is installed. Above you configure as RGB by default. I think if you want to allow your class to have an option for a different strip type, you should take it in as a parameter in the constructor.

Reply:
done